File: packages/templates/queues/rabbitmq-node/src/queue/rabbitmq.client.ts
/**
 * RabbitMQ AMQP Client & Message Publisher / Consumer
 */
export class RabbitMqClient {
  private url: string;

  constructor(url = process.env.RABBITMQ_URL || 'amqp://guest:guest@localhost:5672') {
    this.url = url;
  }

  async publish(queueName: string, message: Record<string, unknown>): Promise<void> {
    console.log(`[RabbitMQ] Publishing to "${queueName}" ->`, JSON.stringify(message));
    // Implementation can hook amqplib.connect(this.url) here
  }

  async consume(queueName: string, onMessage: (msg: unknown) => void): Promise<void> {
    console.log(`[RabbitMQ] Subscribing to queue "${queueName}" on ${this.url}`);
  }
}

export const rabbitMq = new RabbitMqClient();

File: packages/templates/queues/rabbitmq-python/app/queue/rabbitmq_client.py
import os
import json

class RabbitMQClient:
    def __init__(self, amqp_url: str = None):
        self.amqp_url = amqp_url or os.getenv("RABBITMQ_URL", "amqp://guest:guest@localhost:5672/")

    def publish(self, queue_name: str, payload: dict):
        print(f"[RabbitMQ] Publishing message to {queue_name}: {json.dumps(payload)}")

    def consume(self, queue_name: str, callback):
        print(f"[RabbitMQ] Listening on queue: {queue_name}")

rabbitmq_client = RabbitMQClient()
